server/src/test/java/org/elasticsearch/cluster/routing/allocation/BalanceConfigurationTests.java

```diff
@@ -318,155 +312,4 @@ public void testPersistedSettings() {
         assertThat(allocator.getShardBalance(), Matchers.equalTo(0.1f));
         assertThat(allocator.getThreshold(), Matchers.equalTo(3.0f));
     }
-
-    public void testNoRebalanceOnPrimaryOverload() {
```

Comment:
I don't think this test was testing anything much. It sets up a specific routing table and then runs the balancer, but it only asserts that all the _assigned_ shards remain `STARTED` which is unsurprising and surely covered elsewhere. It seems to be expecting that the balancer would spread out the primaries but I don't think it actually does.